node/src/types/validator_matrix.rs

```diff
@@ -122,17 +130,21 @@ impl ValidatorMatrix {
             .write()
             .expect("poisoned lock on validator matrix");
         let is_new = guard.insert(era_id, validators).is_none();
-        if guard.len() > MAX_VALIDATOR_MATRIX_ENTRIES {
-            // Safe to unwrap because we check above that we have sufficient entries.
-            let median_key = guard
-                .keys()
-                .nth(MAX_VALIDATOR_MATRIX_ENTRIES / 2)
-                .copied()
-                .unwrap();
-            guard.remove(&median_key);
-            return median_key != era_id;
+        if guard.len() <= MAX_VALIDATOR_MATRIX_ENTRIES || self.retrograde_latch.is_none() {
+            return is_new;
+        }
+        let latch_era = self.retrograde_latch.unwrap_or_default(); // none checked above
```

Comment:
Generally we try to avoid any unwraps where possible. To get rid of the `unwrap_or_default`, we can do:
```suggestion
        let latch_era = if let Some(era) = self.retrograde_latch.as_ref() {
            *era
        } else {
            return is_new;
        }
        if guard.len() <= MAX_VALIDATOR_MATRIX_ENTRIES {
            return is_new;
        }
```

Reply:
While I don't agree that we should avoid `unwrap_or_default()` generally, I do slightly prefer the suggested change as it seems a little more straighforward to me.

Reply:
while it is true we avoid `unwrap` as much as possible, we use `unwrap_or_default` where safe quite a bit. 

as to the rest of it, I prefer early return and DRY; there's no reason to unpack the era if we're not at `MAX_VALIDATOR_MATRIX_ENTRIES` similarly, if the latch has not been set yet, there's no reason to do any further work...so get out. 

but, i'll make the change to move this along.